src/pretix/control/forms/event.py

```diff
@@ -985,10 +985,28 @@ class MailSettingsForm(SettingsForm):
         widget=I18nTextarea,
     )
     smtp_use_custom = forms.BooleanField(
-        label=_("Use custom SMTP server"),
-        help_text=_("All mail related to your event will be sent over the smtp server specified by you."),
+        label=_("Use Custom Email"),
+        help_text=_("All mail related to your event will be sent over your specified email gateway."),
         required=False
     )
+    send_grid_api_key = forms.CharField(
+        label=_("Sendgrid Token"),
+        required=True,
+        widget=forms.TextInput(attrs={'placeholder': 'SG.xxxxxxxx'})
+    )
+
+    smtp_select = [
+        
+        ('sendgrid', _("SendGrid")),
+        ('smtp', _("SMTP"))]
+
+
+    email_vendor = forms.ChoiceField(
+        label=_(""),
+        required=True,
+        widget=forms.RadioSelect,
+        choices=smtp_select
```

Comment:
For value of `choices`, please use Django [`TextChoices`](https://docs.djangoproject.com/en/5.0/ref/models/fields/#enumeration-types).

Instead of  defining choices nested in a model, please move it outside, so that it can be easily imported to use in other places.